api/include/opentelemetry/common/macros.h

```diff
@@ -7,9 +7,7 @@
 // GCC 9 has likely attribute but do not support declare it at the beginning of statement
 #  if defined(__has_cpp_attribute) && (defined(__clang__) || !defined(__GNUC__) || __GNUC__ > 9)
 #    if __has_cpp_attribute(likely)
-#      define OPENTELEMETRY_LIKELY_IF(...) \
-        if (__VA_ARGS__)                   \
-        [[likely]]
+#      define OPENTELEMETRY_LIKELY_IF(...) if (__VA_ARGS__) [[likely]]
```

Comment:
I agree the suggestion totally make sense and I was considering rewrote it. But since it is not introduced here and used somewhere else, perhaps make the change in a different PR? Then that PR will be straightforward and could be merged before the current PR.

Reply:
One more thing to add. The macro tries to prefer the likely attribute, `if (cond) [likely]` which cannot be defined like the same as `likely/unlikely` macros. Perhaps this is the reason to define it with if condition check.

Reply:
> One more thing to add. The macro tries to prefer the likely attribute, `if (cond) [likely]` which cannot be defined like the same as `likely/unlikely` macros. Perhaps this is the reason to define it with if condition check.

Yes, it's.

Reply:
Maybe just:

```
#define OPENTELEMETRY_LIKELY [[likely]]
```

so this can be used in if, but also in switch case, since this attribute is per branch (not expression as I assumed)

To address separately anyway, setting this issue resolved.

Reply:
> Maybe just:
> 
> ```
> #define OPENTELEMETRY_LIKELY [[likely]]
> ```
> 
> so this can be used in if, but also in switch case, since this attribute is per branch (not expression as I assumed)
> 
> To address separately anyway, setting this issue resolved.

How can we adapt the likely/unlikely in GCC/Clang through this approach? We hope to maintain consistency in the usage of both C++ attributes and traditional compiler built-in macros.

Reply:
BTW, created a new issue to track this.

https://github.com/open-telemetry/opentelemetry-cpp/issues/2098